catroid/src/test/java/org/catrobat/catroid/test/physics/clone/ClonePhysicsBricksWithFormulaTest.java

```diff
@@ -0,0 +1,109 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+
+import static junit.framework.Assert.assertNotSame;
+
+import static org.junit.Assert.assertSame;
+
+@RunWith(Parameterized.class)
+public class ClonePhysicsBricksWithFormulaTest {
+
+	private static final int BRICK_FORMULA_VALUE = 0;
+	private static final String BRICK_INVALID_FORMULA_VALUE = "1";
+	private static final String CLONE_BRICK_FORMULA_VALUE = "2";
+
+	@Parameterized.Parameters(name = "{0}")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+				{"SetBounceBrick", new SetBounceBrick(new Formula(BRICK_FORMULA_VALUE)), Brick.BrickField.PHYSICS_BOUNCE_FACTOR},
+				{"SetFrictionBrick", new SetFrictionBrick(new Formula(BRICK_FORMULA_VALUE)), Brick.BrickField.PHYSICS_FRICTION},
+				{"SetGravityBrickX", new SetGravityBrick(new Formula(BRICK_FORMULA_VALUE), new Formula(BRICK_INVALID_FORMULA_VALUE)), Brick.BrickField.PHYSICS_GRAVITY_X},
+				{"SetGravityBrickY", new SetGravityBrick(new Formula(BRICK_INVALID_FORMULA_VALUE), new Formula(BRICK_FORMULA_VALUE)), Brick.BrickField.PHYSICS_GRAVITY_Y},
+				{"SetMassBrick", new SetMassBrick(new Formula(BRICK_FORMULA_VALUE)), Brick.BrickField.PHYSICS_MASS},
+				{"SetVelocityBrickX", new SetVelocityBrick(new Formula(BRICK_FORMULA_VALUE), new Formula(BRICK_INVALID_FORMULA_VALUE)), Brick.BrickField.PHYSICS_VELOCITY_X},
+				{"SetVelocityBrickY", new SetVelocityBrick(new Formula(BRICK_INVALID_FORMULA_VALUE), new Formula(BRICK_FORMULA_VALUE)), Brick.BrickField.PHYSICS_VELOCITY_Y},
+				{"TurnRightSpeedBrick", new TurnRightSpeedBrick(new Formula(BRICK_FORMULA_VALUE)), Brick.BrickField.PHYSICS_TURN_RIGHT_SPEED},
+				{"TurnLeftSpeedBrick", new TurnLeftSpeedBrick(new Formula(BRICK_FORMULA_VALUE)), Brick.BrickField.PHYSICS_TURN_LEFT_SPEED},
+		});
+	}
+
+	@Parameterized.Parameter
+	public String name;
+
+	@Parameterized.Parameter(1)
+	public FormulaBrick brick;
+
+	@Parameterized.Parameter(2)
+	public Brick.BrickField brickField;
+
+	private Sprite sprite = new SingleSprite("testSprite");
+	private Formula brickFormula;
+	private Formula cloneBrickFormula;
+
+	@Before
+	public void setUp() throws CloneNotSupportedException {
+		FormulaBrick cloneBrick = (FormulaBrick) brick.clone();
+		brickFormula = brick.getFormulaWithBrickField(brickField);
+		cloneBrickFormula = cloneBrick.getFormulaWithBrickField(brickField);
+	}
+
+	@Test
+	public void testChangeBrickField() throws InterpretationException {
+		cloneBrickFormula.setRoot(new FormulaElement(FormulaElement.ElementType.NUMBER, CLONE_BRICK_FORMULA_VALUE, null));
+		assertNotSame(brickFormula.interpretInteger(sprite), cloneBrickFormula.interpretInteger(sprite));
+	}
+
+	@Test
+	public void testBrickFieldValidValue() throws InterpretationException {
+		assertSame(BRICK_FORMULA_VALUE, brickFormula.interpretInteger(sprite));
+		assertSame(BRICK_FORMULA_VALUE, cloneBrickFormula.interpretInteger(sprite));
+	}
```

Comment:
if you delete second line here, you can add the assert from testBrickFieldEquals to do the same thing but in one
`assertSame(brickFormula.interpretInteger(sprite), cloneBrickFormula.interpretInteger(sprite));`
So you are interpreting the correct value first, and then check the cloned one